src/python/WMCore/WorkQueue/WMBSHelper.py

```diff
@@ -511,8 +511,13 @@ def _createDatasetSubscriptionsInDBSBuffer(self):
         existingTransaction = self.beginTransaction()
 
         subInfo = self.wmSpec.getSubscriptionInformation()
+
         for dataset in subInfo:
-            dbsDataset = DBSBufferDataset(path=dataset)
```

Comment:
Are you sure this is the correct place for fetching and setting this parentage info? To me it seems it will only add IF the workflow was assigned with data subscriptions set.